### PR TITLE
update the words below the notes

### PR DESCRIPTION
**Pull request by purerosefallen**
QQ图片20191104210019

Only TJA is supported at this moment..

### CODE REVIEW

**Comment by bui** [comment]
appears to be unfinished. example: 3rd note in 虹色・夢色・太鼓色

AC:
image

this PR:
image

File: public/src/js/strings.js [hunk 1]
```diff
@@ -190,6 +206,22 @@ function StringsEn(){
 		daiDrumroll: "DRUM ROLLー!!",
 		balloon: "Balloon"
 	}
+	this.ex_note = {
+		don: [
+			"Don",
+			"Don"
```
Comment (LoveEevee) [inline public/src/js/strings.js]: Both `ド` and `コ` translate to `Do` on Switch version

File: public/src/js/strings.js [hunk 2]
```diff
@@ -610,6 +674,22 @@ function StringsKo(){
 		daiDrumroll: "연타(대)ー!!",
 		balloon: "풍선"
 	}
+	this.ex_note = {
+		don: [
+			"쿵",
+			"쿵"
```
Comment (LoveEevee) [inline public/src/js/strings.js]: Both are "쿠" on Switch version

**Comment by purerosefallen** [comment]
Do you have any idea when do and ko would appear?

**Comment by LoveEevee** [comment]
> Do you have any idea when do and ko would appear?

On arcade or console they are baked into the fumen format itself, for simulators you have to guess. I opened Nijiiro Yumeiro Taikoiro on TJAPlayer 3 and it shows ドン for the third note, so in that specific example the code is doing it correctly in my opinion.

**Comment by purerosefallen** [comment]
@LoveEevee So I think the TJA format should somehow be changed, using 3 different characters as do ko don.
There would be many odd examples. Like Nesin Amatias, the April fool Song of 2019, using Don Ko Do instead.
However, doing this change would have to remake every chart, which is a very hard work.

**Comment by LoveEevee** [comment]
@purerosefallen [#SENOTECHANGE is already a thing](https://github.com/bui/taiko-web/wiki/TJA-format#senotechange-), though it is not used often

**Comment by LoveEevee** [comment]
```
0000000000000011,
1110000000000000,
```
image

This should be ドコドコドン

**Comment by LoveEevee** [comment]
```
1111100000000011,
1110000000000022,
1110000000000000,
```
Currently it writes this:
`ドコドコドン　　ドドドン　　かかドコドン`
It should write this instead:
`ドドドドドン　　ドコドン　　かかドドドン`

**Comment by purerosefallen** [comment]
> ```
> 1111100000000011,
> 1000000000000022,
> 1110000011110000,
> ```
> 
> Currently it writes this:
> `ドコドコドン　　ドドドン　　カカドコドン　　ドコドドン`
> It should write this instead:
> `ドドドドドン　　ドコドン　　カカドドドン　　ドドドドン`
> 
> Edit: I am unsure about the first pattern though, some songs have `ドコドコドン` and others have `ドドドドドン`


Yeah, in the original Taiko, the odd number of dons would always become `ドコドコドン`. However, it varies when it comes to even number of dons. Some songs give `ドコドドン` while others give `ドドドドン`.

In this PR, the logic is as follows. If don continuously appears in the last part of a group of notes, the `コ` appears in every 2nd, 4th ... note. I'm still not sure how Namco think of the note lyrics.

**Comment by TonalidadeHidrica** [comment]
Excuse me for cutting in; actually, the lyrics under the notes are set completely manually, though there are some guidelines in which the rhythm patterns are intuitively translated to.

**Pull request merged**